### PR TITLE
Event Sourcing for Grabbables

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

- [x] Fixes netcode for authority transfer
- [x] Implements Event Sourcing for Grabbables
- [x] Renames equippables to grabbables
- [x] Fixes testing configuration problems

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/engine/src/interaction/components/GrabbableComponent.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+/**
+ * GrabberComponent
+ * - Allows an entity to grab a GrabbableComponent
+ */
+export const GrabberComponent = defineComponent({
+  name: 'GrabberComponent',
+
+  onInit(entity) {
+    return {
+      grabbedEntity: null as Entity | null
```
Comment (speigg) [inline packages/engine/src/interaction/components/GrabbableComponent.ts:93]: Does this mean only one object can be grabbed at a time? Why limit it in this way?

**Review by speigg — CHANGES_REQUESTED** [review]

**Comment by speigg** [comment]
When a new client joins, the grabbable does not appear in the current position.

**Comment by HexaField** [comment]
> When a new client joins, the grabbable does not appear in the current position.

you'll need to reinit your db, this is due to it spawning as a fixed rigidbody and not dynamic. the problem with this then though, is it falls through the floor because the skybase loads after it

File: packages/engine/src/interaction/components/GrabbableComponent.ts (hunk 1 above)
Comment (HexaField) [reply packages/engine/src/interaction/components/GrabbableComponent.ts:93]: for now, yeah. this PR is just refactoring this functionality, not extending it

**Pull request merged**